doc/sphinx/language/gallina-extensions.rst

```diff
@@ -591,29 +597,85 @@ This example emphasizes what the printing settings offer.
+
+   *  If :n:`@fixannot` is not specified, :cmd:`Function`
+      defines the nonrecursive function :token:`ident` as if it was declared with
+      :cmd:`Definition`. In addition, the following are defined:
+
+       + :token:`ident`\ ``_rect``, :token:`ident`\ ``_rec`` and :token:`ident`\ ``_ind``,
+         which reflect the pattern matching structure of :token:`term` (see :cmd:`Inductive`);
+       + The inductive :n:`R_@ident` corresponding to the graph of :token:`ident` (silently);
+       + :token:`ident`\ ``_complete`` and :token:`ident`\ ``_correct`` which
+         are inversion information linking the function and its graph.
+
+   *  If :n:`{ struct ... }` is specified, :cmd:`Function`
+      defines the structural recursive function :token:`ident` as if it was declared
+      with :cmd:`Fixpoint`. In addition, the following are defined:
+
+       + The same objects as above;
+       + The fixpoint equation of :token:`ident`: :n:`@ident`\_equation`.
+
+       .. todo: these idents above are the not the one in { struct ident }, right?
+          AFAIK we don't describe the ident for struct anywhere
+
+   *  If :n:`{ measure ... }` or :n:`{ wf ... }` are specified, :cmd:`Function`
+      defines a recursive function by well-founded recursion. The module ``Recdef``
+      of the standard library must be loaded for this feature.
+
+       + :n:`{measure @one_term {? @ident } {? @one_term } }`\: where :n:`@ident` is the decreasing argument
+         and :n:`@one_term__1` is a function from the type of :n:`@ident` to :g:`nat` for which
+         the decreasing argument decreases (for the :g:`lt` order on :g:`nat`)
+         for each recursive call of :n:`@one_term__1`. Parameters of the function are
+         bound in :n:`@one_term`.
+       + :n:`{wf @one_term @ident }`\: where :n:`@ident` is the decreasing argument and
+         :n:`@one_term` is an ordering relation on the type of :n:`@ident` (i.e. of type
+         `T`\ :math:`_{\sf ident}` → `T`\ :math:`_{\sf ident}` → ``Prop``) for which the decreasing argument
+         decreases for each recursive call of :n:`@one_term`. The order must be well-founded.
+         Parameters of the function are bound in :n:`@one_term`.
+
+      If the annotation is ``measure`` or ``fw``, the user is left with some proof
+      obligations that will be used to define the function. These proofs
+      are: proofs that each recursive call is actually decreasing with
+      respect to the given criteria, and (if the criteria is `wf`) a proof
+      that the ordering relation is well-founded. Once proof obligations are
+      discharged, the following objects are defined:
+
+        + The same objects as with the struct;
+        + The lemma `ident`\ :math:`_{\sf tcc}` which collects all proof obligations in one
+          property;
+        + The lemmas `ident`\ :math:`_{\sf terminate}` and `ident`\ :math:`_{\sf F}` which is needed to be inlined
```

Comment:
```suggestion
        + The lemmas :n:`@ident`\ ``_terminate`` and :n:`@ident`\ ``_F`` which is needed to be inlined
```

Reply:
Got it.  I changed "which is needed to be inlined" to "which need to be inlined".  Is that good or is "which will be inlined" [by Coq] better?

Reply:
I'd say the latter, but I'm not quite sure.

Reply:
Made it "will be".